docs/rulesets.md

```diff
@@ -0,0 +1,168 @@
+      <td><code>boolean</code></td>
+      <td>should the rule be enabled by default?</td>
+    </tr>        
+    <tr>
+      <td>when</td>
+      <td><code><a href="#when">When</a></code></td>
+      <td>A filter object to narrow down the selected items.</td>
+    </tr>
+  </tbody>
+</table>
+
+### Then
+
+<table>
+  <thead>
+    <tr>
+      <th>Field</th>
+      <th>Type</th>
+      <th>Description</th>
+    </tr>
+  </thead>
+  <tbody>    
+    <tr>
+      <td>function</td>
+      <td><code>string</code></td>
+      <td><b>Required</b> Name of the function to run. Note: to use a function you must first register it (<code>spectral.addFunctions(...)</code>). Please check out the default common functions.</td>
+    </tr>
+    <tr>
+      <td>field</td>
+      <td><code>string</code></td>
+      <td>Name of the field to narrow by. Given this field, the function will receive value of that field.</td>
+    </tr>    
+    <tr>
+      <td>functionOptions</td>
+      <td><code>Object</code></td>
+      <td>Options passed to the function. Specific to the function in use.</td>
+    </tr>
+  </tbody>
+</table>
+
+### When
+
+<table>
+  <thead>
+    <tr>
+      <th>Field</th>
+      <th>Type</th>
+      <th>Description</th>
+    </tr>
+  </thead>
+  <tbody>    
+    <tr>
+      <td>field</td>
+      <td><code>string</code></td>
+      <td><b>Required</b> The <code>path.to.prop</code> to field, or special <code>@key</code> value to target keys for matched <code>given</code> object. <br/><br/> Example: if the target object is an oas object and given = <code>$..responses[*]</code>, then <code>@key</code> would be the response code (200, 400, etc)</td>
+    </tr>
+    <tr>
+      <td>patterns</td>
+      <td><code>string</code></td>
+      <td>A regex pattern</td>
+    </tr>
+  </tbody>
+</table>
+
+## Configuring rulesets via CLI
+
+```bash
+spectral lint foo.yaml --ruleset=path/to/acme-company-ruleset.yaml
+```
+
+### Ruleset validation
+
+We use JSON Schema & AJV to validate your rulesets file and help you spot issues early.
+
+**Example output**
+
+```bash
+spectral lint some-oas.yaml --ruleset acme-company.json
+
+Reading ruleset
+
+/rules/rule-without-given-nor-them 	 should have required property 'given' 
+/rules/rule-without-given-nor-them 	 should have required property 'then' 
+/rules/rule-with-invalid-enum/severity 	 should be number 
+/rules/rule-with-invalid-enum/severity 	 should be equal to one of the allowed values 
+/rules/rule-with-invalid-enum/type 	 should be equal to one of the allowed values 
+```
+
+**Note for developers**
+
+Supporting YAML and JSON file validation doesn't come free. 
+We need to maintain schema files that mirror IRule and IRuleset types (see `/src/meta/*.schema.json`).
+Ideally, we would have a script that converts TS type to JSON Schema and keeps the meta files up to date. As of now we have a helper that partially automates the work.
+
+Execute `yarn schema.  update` to recreate the `/src/meta/rule.schema.json`.
```

Comment:
This yarn schema.   update still looks funny to me, what is it?

Reply:
It's a helper. It will take the `IRule` type from `types.ts` file and automatically update `rule.schema.json` file (the JSON Schema represantation I use to validate rulesets). It can be done manually but this script is pretty helpful. There shouldn't be a whitespace here though.